src/providers/google/helpers.ts

```diff
@@ -171,3 +173,245 @@ export class GmailHelpers {
     return { name: "", email: "" };
   }
 }
+
+export class GoogleDriveHelpers {
+  static async getFile(
+    drive: drive_v3.Drive,
+    fileId: string
+  ): Promise<drive_v3.Schema$File> {
+    try {
+      const res = await drive.files.get({
+        fileId: fileId,
+        fields: 'id, name, mimeType, size, webViewLink, createdTime, modifiedTime, thumbnailLink'
+      });
+      return res.data;
+    } catch (error) {
+      console.error("Error getting file:", error);
+      throw error;
+    }
+  }
+
+  static async getFileSize(
+    drive: drive_v3.Drive,
+    fileId: string
+  ): Promise<number | undefined> {
+    const file = await this.getFile(drive, fileId);
+
+    if (file.size) {
+      // For non-Google docs (like PDF, image)
+      return parseInt(file.size);
+    } else if (file.mimeType && file.mimeType.startsWith("application/vnd.google-apps.")) {
+      // For Google Docs, export the file as plain text to estimate size
+      const exportedFile = await drive.files.export(
+        { fileId: fileId, mimeType: "text/plain" }, 
+        { responseType: "arraybuffer" }
+      );
+      return Buffer.byteLength(exportedFile.data as ArrayBuffer);
+    } else {
+      return undefined;
+    }
+  }
+
+  static async extractIndexableText(
```

Comment:
What happened to using `indexableText`?

We don't want to download files if we don't have to.

Reply:
> What happened to using `indexableText`?
> 
> We don't want to download files if we don't have to.

Doing some tests

Reply:
> What happened to using `indexableText`?
> 
> We don't want to download files if we don't have to.

I've added extracting `indexableText` if available. Very rare to check indexable text and it takes some time to index documents for GDrive